src/mrpro/data/SpatialDimension.py

```diff
@@ -101,3 +102,136 @@ def zyx(self) -> tuple[T, T, T]:
     def __str__(self) -> str:
         """Return a string representation of the SpatialDimension."""
         return f'z={self.z}, y={self.y}, x={self.x}'
+
+    def __getitem__(
+        self: SpatialDimension[torch.Tensor], idx: typeing_utils.IndexerType
+    ) -> SpatialDimension[torch.Tensor]:
+        """Get SpatialDimension item."""
+        if not all(isinstance(el, torch.Tensor) for el in self.zyx):
+            raise IndexError('Cannot index SpatialDimension with non-tensor')
+        return SpatialDimension(self.z[idx], self.y[idx], self.x[idx])
+
+    def __setitem__(self: SpatialDimension[torch.Tensor], idx: typeing_utils.IndexerType, other: SpatialDimension):
+        """Set SpatialDimension item."""
+        if not all(isinstance(el, torch.Tensor) for el in self.zyx):
+            raise IndexError('Cannot index SpatialDimension with non-tensor')
+        self.z[idx] = other.z
+        self.y[idx] = other.y
+        self.x[idx] = other.x
+
+    @overload
+    def __mul__(self: SpatialDimension[int], other: int | SpatialDimension[int]) -> SpatialDimension[int]: ...
+    @overload
+    def __mul__(self: SpatialDimension[int], other: float | SpatialDimension[float]) -> SpatialDimension[float]: ...
+    @overload
+    def __mul__(self: SpatialDimension[float], other: float | SpatialDimension[float]) -> SpatialDimension[float]: ...
+
+    @overload
+    def __mul__(
+        self: SpatialDimension[torch.Tensor], other: float | T | SpatialDimension
```

Comment:
why do we need the an explicit `float` here? Shouldn't `float` be covered by `T` already?